cdc/generation.cc

```diff
@@ -842,15 +953,15 @@ future<> generation_service::handle_cdc_generation(std::optional<db_clock::time_
     }
 }
 
-void generation_service::async_handle_cdc_generation(db_clock::time_point ts) {
+void generation_service::async_handle_cdc_generation(db_clock::time_point ts, gms::inet_address source) {
     assert_shard_zero(__PRETTY_FUNCTION__);
 
-    (void)(([] (db_clock::time_point ts, shared_ptr<generation_service> svc) -> future<> {
+    (void)(([] (db_clock::time_point ts, gms::inet_address source, shared_ptr<generation_service> svc) -> future<> {
         while (true) {
             co_await sleep_abortable(std::chrono::seconds(5), svc->_abort_src);
```

Comment:
It's a preexisting condition, but this snippet will result in annoying warnings about discarded exceptional futures if this background job fails with `sleep_aborted` exception. It would be best to just keep the handle to this future somewhere, but I don't see a good place for it, so maybe at least add a `.handle_exception()` leaf at the end to eat the exception and ignore it.